utils/tests_fetcher.py

```diff
@@ -512,15 +513,40 @@ def print_tree_deps_of(module, all_edges=None):
         print(line[0])
 
 
+def init_test_examples_dependencies():
+    """
+    The test examples do not import from the examples (which are just scripts, not modules) so we need som extra
+    care initializing the dependency map there.
+    """
+    test_example_deps = {}
+    all_examples = []
+    for framework in ["flax", "pytorch", "tensorflow"]:
+        test_files = list((PATH_TO_EXAMPLES / framework).glob("test_*.py"))
+        all_examples.extend(test_files)
```

Comment:
Is the reason to include example test files (and return `all_examples`) only for testing purpose in `test_tests_fetcher.py`, and not really necessary for the line (below) `all_modules = list(PATH_TO_TRANFORMERS.glob("**/*.py")) + list(PATH_TO_TESTS.glob("**/*.py")) + examples`? (The `examples` part). See my comment there.

Reply:
No it's just there is a custom logic with some files in the example folder being test files with weird imports and other being example files.